indexer/src/main/java/au/org/aodn/esindexer/service/IndexerServiceImpl.java

```diff
@@ -184,12 +187,21 @@ protected StacCollectionModel getMappedMetadataValues(String metadataValues) thr
         stacCollectionModel.getSummaries().setScore(score);
 
         // parameter vocabs
-        List<String> mappedParameterVocabsFromGcmdKeywords = gcmdKeywordUtils.getMappedParameterVocabsFromGcmdKeywords(stacCollectionModel.getThemes());
-        List<String> processedParameterVocabs = vocabService.extractVocabLabelsFromThemes(stacCollectionModel.getThemes(), AppConstants.AODN_DISCOVERY_PARAMETER_VOCABS);
+        Set<String> mappedParameterLabels = new HashSet<>();
+        List<String> processedParameterVocabs = vocabService.extractVocabLabelsFromThemes(
+                stacCollectionModel.getThemes(), AppConstants.AODN_DISCOVERY_PARAMETER_VOCABS
+        );
 
         if (!processedParameterVocabs.isEmpty()) {
-            stacCollectionModel.getSummaries().setParameterVocabs(Stream.concat(mappedParameterVocabsFromGcmdKeywords.stream(), processedParameterVocabs.stream()).distinct().collect(Collectors.toList()));
+            mappedParameterLabels.addAll(processedParameterVocabs);
```

Comment:
In the GCMDKeyworkUtils, there is a readResourceFile, I think you do not need to code this but use the FileUtils from apache common and/or ResourceUtils from springframwork

The function loadCsvToMap have similar function in apache common, https://www.baeldung.com/apache-commons-csv you can use this or google a bit without this import

Also please add comment in this class to explain where the csv comes from and how to update it etc

Reply:
ahh missed this

Reply:
Need change to this Utils file, miss to checkin?